src/lib/IDE/Intellij.py

```diff
@@ -0,0 +1,170 @@
+            if "No matches found" in str(exc):
+                return False
+            else:
+                raise Exception(exc)
+
+    def open_mta_perspective(self):
+        """
+        Opens MTA perspective in IntelliJ
+        """
+        if self.is_open_mta_perspective():
+            logging.info("MTA perspective is already opened !")
+            return
+        else:
+            # Click on the MTA tab in left sidebar
+            self.click_element(locator_type="image", locator="mta_tab.png")
+
+    def run_simple_analysis(self, project, migration_target, packages=[]):
+        """
+        Runs analysis by adding the project and/or packages passed as argument
+
+        Args:
+            project (str): Full name of project to be analysed
+            migration_target (str): Target technology for migration
+            packages (list): List of packages to be added to analysis
+
+        Returns:
+            None
+
+        Steps:
+            1) Click on MTA Configuration tab
+            2) Right click anywhere and create new configuration
+            3) Type project name in source
+            4) Provide mta cli path, if not present
+            5) Select the target technology
+            6) Right click on config name and run
+            7) Confirm analysis has started
+        """
+        try:
+            self.click_element(locator_type="image", locator="console_opened.png")
+            self.press_keys("shift", "esc")
+        except Exception as exc:
+            try:
+                self.click_element(locator_type="image", locator="console_opened_alt.png")
+                self.press_keys("shift", "esc")
+            except Exception:
+                logging.debug("Console is already closed !")
+            logging.debug("Console is not opened ! {}".format(str(exc)))
+        config_create_region = self.two_coordinate_locator(
+            locator_type="point",
+            x_coordinate=110,
+            y_coordinate=370,
+        )
+        self.click(config_create_region)
+        self.click(action="right_click")
+        self.press_keys("down")
+        self.press_keys("enter")
+        time.sleep(3)
+        self.wait_find_element(locator_type="image", locator="mta_config_page_opened.png")
+        # Region defining the configuration name
+        config_name_region = self.define_region(778, 248, 1125, 284)
+        add_project_locator = self.image_locator("add_project_button.png")
+        add_project_buttons = self.find_elements(add_project_locator)
+        # Provide mta cli path
+        try:
+            # self.click_element(locator_type="image", locator="mta_cli_input.png")
+            self.click(add_project_buttons[0])
+            time.sleep(2)
+            for letter in config_data["mta_cli_path"]:
+                self.type_text(letter)
+            self.press_keys("enter")
+        except Exception as exc:
+            logging.debug("MTA cli path is already present ! {}".format(exc))
```

Comment:
This method is very big.
We can create fixtures or utility methods for smaller tasks like :  


fill_cli_path()
fill_target()
wait_for_analysis_to_complete()  etc

Reply:
noted, will create future PR's for this

Reply:
jira ticket created: https://issues.redhat.com/browse/WINDUP-3662